src/OrbitGl/include/OrbitGl/BatchRenderGroup.h

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_BATCH_RENDER_GROUP_H_
+#define ORBIT_GL_BATCH_RENDER_GROUP_H_
+
+#include <functional>
+#include <string>
+
+#include "absl/container/flat_hash_map.h"
+
+namespace orbit_gl {
+
+// A BatchRenderGroupId identifies a group of primitives that can be rendered together in a single
+// draw call. It is conceptually similar to a layer z-value, but it also includes a name that can be
+// used to manually separate elements into different groups. It is used by the Batcher and
+// TextRenderer to group elements into draw calls.
+struct BatchRenderGroupId {
+  static const std::string kGlobalGroup;
+
+  std::string name = kGlobalGroup;
+  float layer = 0;
+
+  friend bool operator==(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return lhs.layer == rhs.layer && lhs.name == rhs.name;
+  }
+
+  friend bool operator!=(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return !(lhs == rhs);
+  }
+
+  friend bool operator<(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return lhs.layer < rhs.layer || (lhs.layer == rhs.layer && lhs.name < rhs.name);
+  }
+
+  friend bool operator<=(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return lhs < rhs || lhs == rhs;
+  }
+
+  friend bool operator>(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return !(lhs <= rhs);
+  }
+
+  friend bool operator>=(const BatchRenderGroupId& lhs, const BatchRenderGroupId& rhs) {
+    return !(lhs < rhs);
+  }
+
+  BatchRenderGroupId& operator=(const BatchRenderGroupId& rhs) = default;
+  BatchRenderGroupId(const BatchRenderGroupId& rhs) = default;
+  BatchRenderGroupId(BatchRenderGroupId&& rhs) = default;
```

Comment:
You don't need to define those. In fact not defining the move assignment operator, but defining the copy assignment operator, disables the implicit move assignment operator. It's better just to define none of these.